src/python/platforms/fuchsia/undercoat.py

```diff
@@ -54,12 +54,14 @@ class UndercoatError(Exception):
 
 def undercoat_api_command(*args):
   """Make an API call to the undercoat binary."""
+  logs.log('Running undercoat command %s' % (args,))
   bundle_dir = environment.get_value('FUCHSIA_RESOURCES_DIR')
   undercoat_path = os.path.join(bundle_dir, 'undercoat', 'undercoat')
   undercoat = new_process.ProcessRunner(undercoat_path, args)
   # The undercoat log is sent to stderr, which we capture to a tempfile
   with tempfile.TemporaryFile() as undercoat_log:
-    result = undercoat.run_and_wait(stderr=undercoat_log)
+    result = undercoat.run_and_wait(
+        stderr=undercoat_log, extra_env={"TMPDIR": "/tmp"})
```

Comment:
nit: single quotes for strings.

also, let's use bot.fuzzer.utils.get_temp_dir() (or a subdirectory of it). This is scoped to the task though. What implications would this have?

Reply:
Ah, whoops! I keep missing these because I rely on the linter to catch stuff like this.

Hm, I believe being task-scoped means that if a task gets interrupted there will be no way for the next run to clean up after the old stale qemu process/disk images. 

That said, looking at the `utils.get_temp_dir()` implementation it seems to basically be scope-equivalent to `tempfile.gettempdir()` which just resolves to the `TMPDIR` envvar in this case. This means I'm pretty sure it would still have the issue I was trying to address here, which was having the files get cleared out within a single task. If this seems wrong to you, though, I can try to dig deeper into debugging why that would be happening.

Any other alternatives you can think of? Could they live in the build dir perhaps?

Reply:
I think we are looking at different functions :)

This is the one I'm talking about: https://github.com/google/clusterfuzz/blob/00cc394d2e0274537f1bc9bc108cad1fd4aa11c2/src/python/bot/fuzzers/utils.py#L130

If this doesn't work though, you'd need to manage your own dir (and make sure you clean it up at the right places to prevent buildup over time).

Something like `os.path.join(environment.get_value('ROOT_DIR'), 'bot', 'undercoat')`. ROOT_DIR/bot already exists, but you'll need to create the undercoat directory if it does not already exist before passing it here.

Reply:
Hm, yeah, that's where I was looking. I'm assuming `FUZZ_INPUTS_DISK` usually isn't set, so it's just a subdir of wherever `tempfile.gettempdir()` points, no? And then https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir lists "1. The directory named by the TMPDIR environment variable." Sorry if I'm totally misunderstanding here.

For the `ROOT_DIR` route, I'll need to think more about what those "right places" are. Maybe it'd be enough to just clear the directory right after running `fuchsia.undercoat.stop_all()` in build_manager? Any handles still left there wouldn't be useful so should be safe to delete them.

Reply:
FUZZ_INPUTS_DISK is always set for a bot. The fallback is for calling this from outside the context of a bot.

And yeah, `fuchsia.undercoat.stop_all()` seems like a reasonable place. As long as there is something that occasionally clears it, the frequency doesn't have to be too high or fine grained.

Reply:
I couldn't get the `fuzzer.utils` method to work, so I went ahead with the `ROOT_DIR` subdir. LMK if that seems good. Thanks for your help sorting this out!